src/defiat/api.ts

```diff
@@ -660,6 +673,168 @@ export const isAbovePeg = async (
+          rewards.returnValues["tokenAmount"],
+          rewards.returnValues["buybackAmount"],
+          getSymbol(rewards.returnValues["token"], chainid),
+          rewards.transactionHash,
+          "Points Buyback",
+          "Out"
+        ));
+      }
+    });
+
+    const outgoingClaims = await AnyStake.getPastEvents("Claim", {
+      fromBlock: blockNumber - 5760,
+      toBlock: blockNumber,
+    });
+
+    await asyncForEach(outgoingClaims, async (rewards) => {
+      const block = await DeFiat.web3.eth.getBlock(rewards.blockNumber);
+      rewardsData.push(new ProcessedRewards(
+        id++,
+        timeConverter(block.timestamp),
+        rewards.returnValues["amount"],
+        undefined,
+        undefined,
+        rewards.transactionHash,
+        "Claimed Rewards",
+        "Out"
+      ));
+    });
+
+    const outgoingClaimsRegulator = await Regulator.getPastEvents("Claim", {
+      fromBlock: blockNumber - 5760,
+      toBlock: blockNumber,
+    });
+    await asyncForEach(outgoingClaimsRegulator, async (rewards) => {
+      const block = await DeFiat.web3.eth.getBlock(rewards.blockNumber);
+      rewardsData.push(new ProcessedRewards(
+        id++,
+        timeConverter(block.timestamp),
+        rewards.returnValues["amount"],
+        undefined,
+        undefined,
+        rewards.transactionHash,
+        "Claimed Rewards",
+        "Out"
+      ));
+    });
+
+    const incomingTransfers = await DeFiatToken.getPastEvents("Transfer", {
+      fromBlock: blockNumber - 5760,
+      toBlock: blockNumber,
+      filter: { to: Addresses.Vault[chainid] },
+    });
+    await asyncForEach(incomingTransfers, async (rewards) => {
+      const block = await DeFiat.web3.eth.getBlock(rewards.blockNumber);
+      if (rewardsData.find((x) => x.transactionHash === rewards.transactionHash) === undefined) {
+        rewardsData.push(new ProcessedRewards(
+          id++,
+          timeConverter(block.timestamp),
+          rewards.returnValues.value,
+          undefined,
+          undefined,
+          rewards.transactionHash,
+          "Transfer Fee",
+          "In"
+        ));
+      }
+    });
+    rewardsData.sort((one, two) => (one.timestamp > two.timestamp ? -1 : 1));
+    return rewardsData;
+
+  } catch (e) {
+    return [];
+  }
+};
+
+function getSymbol(address: string, chainId: number) {
+  if (Addresses.DeFiat[chainId] === address)
+    return "DFTPv2";
+  const pools: StakingPool[] = Pools[chainId];
+  return pools.find((x) => x.address === address).symbol;
+}
+
+function timeConverter(UNIX_timestamp) {
+  var a = new Date(UNIX_timestamp * 1000);
+  var year = a.getFullYear();
+  var month = a.getMonth() + 1;
+  var date = a.getDate();
+  var hour = a.getHours().toString().padStart(2, '0');
+  var min = a.getMinutes().toString().padStart(2, '0');;
+  var sec = a.getSeconds().toString().padStart(2, '0');;
+  var time = year + '-' + month + '-' + date + ' ' + hour + ':' + min + ':' + sec;
+  return time;
```

Comment:
Some notes from ES6+, `var` is hoisted, so its generally best practice to use `let` if you want to modify the variable after assignment. `const` would work here too, since we are not doing something like:

```
const time = 123;
...
time = 111;

```

Also it's best practice to treat the functions as objects to keep everything ES6 -like

```
const timeConverter = (UNIX_timestamp) => {
...
}
```

Code definitely still runs tho, so not a big deal.